hrreport/CRM/HRReport/Upgrader.php

```diff
@@ -255,4 +234,34 @@ public function upgrade_1403() {
     }
     return TRUE;
   }
+
+  /**
+   * Removes unused absence report
+   *
+   * @return bool
+   */
+  public function upgrade_1404() {
+    $templates = [
+      'CRM_HRReport_Form_Activity_HRAbsence',
+      'CRM_HRReport_Form_Activity_HRAbsenceCalendar',
+      'CRM_HRReport_Form_Activity_HRAbsenceDates'
+    ];
+    $reports = civicrm_api3('OptionValue', 'get', [
+      'return' => ['name'],
+      'option_group_id' => 'report_template',
+      'name' => ['IN' => $templates],
+    ]);
+
+    if ($reports['count'] > 0) {
+      $templateIds = array_column(array_values($reports['values']), 'id');
+      foreach ($templateIds as $templateId) {
+        civicrm_api3('OptionValue', 'delete', ['id' => $templateId]);
+      }
```

Comment:
If you check the `.mgd.php` files you've deleted, you'll see that a report is composed of a `ReportTemplate` (The option value you're deleting here) and a `ReportInstance`. You need to delete the report instance too.

Reply:
Alright. I can see they come in pairs of `Template` and `Instance`. Will remove the instances as well. 🗑